### PR TITLE
Rename the branch command to add-branch.

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Rename branch command to add-branch.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap
3. juju branch                     <- ERROR unrecognized command
3. juju add-branch new     <- success
3. check 'see also' of other branch related commands
4. juju help add-branch

## Documentation changes

This feature is still under development, however any docs referencing juju branch should be updated to use juju add-branch in the develop stream.

### CODE REVIEW

File: cmd/juju/model/addbranch.go [hunk 1]
```diff
@@ -29,60 +28,60 @@ The branch name "master" is reserved for primary model-based settings and is
...
-    juju branch upgrade-postgresql
+    juju add-branch upgrade-postgresql
 
 See also:
     track
     checkout
     commit
     abort
-	diff
+    diff
```
Comment (hmlanigan) [inline cmd/juju/model/addbranch.go]: drive by

File: cmd/juju/model/addbranch.go [hunk 2]
```diff
@@ -6,20 +6,19 @@ package model
...
-	"github.com/juju/juju/core/model"
-
 	"github.com/juju/cmd"
 	"github.com/juju/errors"
 	"github.com/juju/gnuflag"
 
 	"github.com/juju/juju/api/modelgeneration"
 	jujucmd "github.com/juju/juju/cmd"
 	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/core/model"
```
Comment (hmlanigan) [inline cmd/juju/model/addbranch.go]: drive by

File: cmd/juju/model/export_test.go [hunk 3]
```diff
@@ -181,8 +181,8 @@ func NewModelCredentialCommandForTest(modelClient ModelCredentialAPI, cloudClien
 	return modelcmd.Wrap(cmd)
 }
 
-func NewBranchCommandForTest(api BranchCommandAPI, store jujuclient.ClientStore) cmd.Command {
-	cmd := &branchCommand{
+func NewBranchCommandForTest(api AddBranchCommandAPI, store jujuclient.ClientStore) cmd.Command {
```
Comment (hmlanigan) [inline cmd/juju/model/export_test.go]: NewBranchCommandForTest should be NewAddBranchCommandForTest

**Comment by manadart** [comment]
Can we target these to the 2.6 branch?

File: cmd/juju/model/addbranch.go (hunk 2 above)
Comment (manadart) [reply cmd/juju/model/addbranch.go]: Thanks, my bad.

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**